src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -900,6 +910,17 @@ class OpenShiftDSL implements Serializable {
             }
         }
 
+        public boolean exists() throws AbortException {
+            return names().size() > 0
```

Comment:
I was wrong here since this implementation will not work with the following:

```
s = openshift.selector( 'pod/x', 'pod/x' )
s.exists()   // This will always return true with the current PR because it creates a static objectList based selector.
```

Instead, we would need a new method:
```
// New method which always queries for names even if the selector is a static list
private queryNames() throws AbortException {
      ...content of current names(), but no opening      if ( objectList != null )   check
}

public ArrayList<String> names() throws AbortException {
        if ( objectList != null ) {
                 return objectList;
        }
        return queryNames()
}

// Tweak to count's implementation
public int count()  throws AbortException {
        return queryNames().size()
}

public boolean exists()  throws AbortException {
         if ( objectList != null ) {
               // If object names are explicitly given, make sure they *all* exists
               return count() == objectList.length
         }
         return count() > 0 
}

``